db/model/Gdoc/htmlToEnriched.ts

```diff
@@ -86,82 +95,998 @@ export function htmlToSimpleTextBlock(html: string): EnrichedBlockSimpleText {
+                            ],
+                        }
+                    }
+
+                    const listChildren = joinBlockParseResults(
+                        children.map(handleListChildren)
+                    )
+                    return {
+                        errors: listChildren.errors,
+                        content: [
+                            {
+                                type: "numbered-list",
+                                items: listChildren.content,
+                                parseErrors: [],
+                            },
+                        ],
+                    }
+                }
+            )
+            .with(
+                { tagName: P.union("svg", "table", "video") },
+                (): BlockParseResult<EnrichedBlockHtml> => {
+                    return {
+                        errors: [],
+                        content: [
+                            {
+                                type: "html",
+                                value: context.$.html(element) ?? "",
+                                parseErrors: [],
+                            },
+                        ],
+                    }
+                }
+            )
+            .otherwise(() => ({
+                errors: [
+                    {
+                        name: "unhandled html tag found",
+                        details: `Encountered the unhandled tag ${element.tagName}`,
+                    },
+                ],
+                content: [],
+            }))
+        return result
+    } else
+        return {
+            errors: [
+                {
+                    name: "unkown element tag",
+                    details: `type was ${element.type}`,
+                },
+            ],
+            content: [],
+        }
+}
+
+export function cheerioElementsToArchieML(
+    elements: CheerioElement[],
+    context: ParseContext
+): BlockParseResult<ArchieBlockOrWpComponent> {
+    if (!elements || !elements.length)
+        return {
+            errors: [],
+            content: [],
+        }
+    let remainingElements: CheerioElement[] = elements
+    const parsedContent: BlockParseResult<ArchieBlockOrWpComponent>[] = []
+    while (remainingElements.length > 0) {
+        const element = remainingElements[0]
+        if (isWpComponentStart(element) && context.shouldParseWpComponents) {
+            const parseResult = parseWpComponent(remainingElements, context)
+            parsedContent.push(parseResult.result)
+            remainingElements = parseResult.remainingElements
+        } else if (element.type === "comment") {
+            remainingElements = remainingElements.slice(1)
+        } else {
+            const parsed = cheerioToArchieML(element, context)
+            const cleaned = withoutEmptyOrWhitespaceOnlyTextBlocks(parsed)
+            if (cleaned.content.length > 0 || cleaned.errors.length > 0)
+                parsedContent.push(parsed)
```

Comment:
as I understand it, pushing `parsed` or `cleaned` doesn't make a difference right now given the test condition, but it would still feel more natural to me to push `cleaned` instead

Reply:
If there is text content we don't want to remove whitespace - this is why parsed is pushed here. I added a comment to explain the reasoning.